beangulp/testing.py

```diff
@@ -1,73 +1,118 @@
+                yield file_or_dir
 
 
-def sha1sum(filepath):
+def sha1sum(filepath: str) -> str:
+    """Compute hash of given filename"""
     with open(filepath, 'rb') as f:
         return hashlib.sha1(f.read()).hexdigest()
 
 
-def write_expected(f, account, date, name, entries):
-    print(f';; Account: {account:}', file=f)
-    print(f';; Date: {date:%Y-%m-%d}', file=f)
-    print(f';; Name: {name:}', file=f)
-    printer.print_entries(entries, file=f)
+def write_expected(outfile: str,
+                   account: Optional[Account],
+                   date: Optional[datetime.date],
+                   name: Optional[str],
+                   entries: data.Entries):
+    """Produce expected file.
+
+    Args:
+      outfile: The name of the expected filename to write.
+      account: The account name produced by the importer.
+      date: The date of the downloads file, produced by the importer.
+      name: The filename for filing, produced by the importer.
+      entries: The list of entries extracted by the importer.
+    """
+    print(';; Account: {}'.format(account or ''), file=outfile)
+    print(';; Date: {}'.format(date.isoformat() if date else ''), file=outfile)
+    print(';; Name: {}'.format(name or ''), file=outfile)
+    printer.print_entries(entries, file=outfile)
 
 
-def write_expected_file(filepath, *data):
-    with open(filepath, 'w') as f:
-        write_expected(f, *data)
+def write_expected_file(filepath: str, *data):
+    """Writes out the expected file."""
+    with open(filepath, 'w') as expfile:
+        write_expected(expfile, *data)
 
 
-def compare_expected(filepath, *data):
-    with io.StringIO() as f:
-        write_expected(f, *data)
+def compare_expected(filepath: str, *data) -> List[str]:
+    """Compare the expected file with extracted data."""
+    with io.StringIO() as buffer:
+        write_expected(buffer, *data)
         # rewind to the beginning of the stream
-        f.seek(0)
-        lines_imported = f.readlines()
+        buffer.seek(0)
+        lines_imported = buffer.readlines()
 
     try:
-        with open(filepath, 'r') as f:
-            lines_expected = f.readlines()
+        with open(filepath, 'r') as infile:
+            lines_expected = infile.readlines()
     except FileNotFoundError:
         lines_expected = []
 
+    # TODO(blais): Is there a way to ignore whitespace here?
+    # The current printer is producing extra junk at the end of lines.
     diff = difflib.unified_diff(lines_expected, lines_imported,
                                 tofile='expected.beancount',
                                 fromfile='imported.beancount')
     return list(diff)
 
 
-def run_importer(importer, f):
-    account = importer.file_account(f)
-    date = importer.file_date(f)
-    name = importer.file_name(f)
-    entries = extract.extract_from_file(f.name, importer, None, None)
+# TODO(blais): Make the FileMemo type public, so it can be used here.
```

Comment:
I would like to get rid of FileMemo, thus I would not bother.

Reply:
SGTM